### PR TITLE
feat: Add images filter in explore page

### LINKED ISSUE #5756: Search Page: Add search filter "Images" to the left side menu and enable Image and Logo by default
Please add filters for logo and image into the left side bar of the search page as follows:
* Add filter option "Images"
* Add filters "Image" and "Logo" as filter options (enabled means the search only shows outputs of events with images and logos)
* Enable "Image" and "Logo" by default

Also
* Additionally change the filter "Event Call for Speakers" to "Call for Speakers"

Screenshot from 2020-11-25 16-26-48

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #5756 

#### Short description of what this resolves:
add images filter

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] The acceptance, integration, unit tests and linter pass locally with my changes 
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Change to has_images, has_logo

File: app/components/explore/side-bar.js [hunk 1]
```diff
@@ -18,9 +18,9 @@ export default class SideBar extends Component {
   isMapVisible = true;
   eventLocationType = null;
 
-  @computed('category', 'sub_category', 'event_type', 'startDate', 'endDate', 'location', 'ticket_type', 'cfs', 'event_name', 'is_online')
+  @computed('category', 'sub_category', 'event_type', 'startDate', 'endDate', 'location', 'ticket_type', 'cfs', 'event_name', 'is_online', 'has_logo', 'has_images')
```
Comment (iamareebjamal) [inline app/components/explore/side-bar.js:21]: has_image, not has_images

File: app/components/explore/side-bar.js [hunk 2]
```diff
@@ -172,6 +182,8 @@ export default class SideBar extends Component {
       cfs               : null,
       event_name        : null,
       is_online         : null,
+      has_logo    : 'false',
+      has_image   : 'false',
```
Comment (iamareebjamal) [inline app/components/explore/side-bar.js:186]: Why dtrinI believe this has wrong indentation? Please check, I'm reviewing on Mobile

Comment (sachinchauhan2889) [reply app/components/explore/side-bar.js:186]: ohh sorry. i give replace all command. thats why identation got wrong. sorry

**Comment by iamareebjamal** [comment]
On mobile, all filters except CFS and image options are hidden. Please hide them as well. And the filters are pre-selected. Please unselect

**Comment by sachinchauhan2889** [comment]
@iamareebjamal plz review

File: app/components/explore/side-bar.js [hunk 3]
```diff
@@ -172,6 +182,8 @@ export default class SideBar extends Component {
       cfs               : null,
       event_name        : null,
       is_online         : null,
+      has_logo          : null,
+      has_image         : null,
```
Comment (iamareebjamal) [inline app/components/explore/side-bar.js:186]: Still wrong indentation

File: app/controllers/explore.js [hunk 4]
```diff
@@ -4,9 +4,11 @@ import Controller from '@ember/controller';
 
 @classic
 export default class ExploreController extends Controller {
-  queryParams = ['category', 'sub_category', 'event_type', 'start_date', 'end_date', 'location', 'ticket_type', 'cfs', 'event_name', 'is_online'];
+  queryParams = ['category', 'sub_category', 'event_type', 'start_date', 'end_date', 'location', 'ticket_type', 'cfs', 'event_name', 'is_online', 'has_logo', 'has_image'];
   event_name = null;
   is_online = null;
+  has_logo = null;
+  has_image = null;
```
Comment (iamareebjamal) [inline app/controllers/explore.js:11]: Indentation

File: app/routes/explore.js [hunk 5]
```diff
@@ -79,6 +79,20 @@ export default class ExploreRoute extends Route {
         val  : params.is_online
       });
     }
+    if (!params.has_image) {
+      filterOptions.push({
+        name : 'original-image-url',
+        op   : 'ne',
+        val  : null
+      });
```
Comment (maze-runnar) [inline app/routes/explore.js:87]: the condition used inside is for checking if it has a image. And it is named like "not has_image".

File: app/routes/explore.js [hunk 6]
```diff
@@ -79,6 +79,20 @@ export default class ExploreRoute extends Route {
...
+    }
+    if (!params.has_logo) {
+      filterOptions.push({
+        name : 'logo-url',
+        op   : 'ne',
+        val  : null
+      });
+    }
```
Comment (maze-runnar) [inline app/routes/explore.js:95]: same here

File: app/routes/explore.js (hunk 5 above)
Comment (sachinchauhan2889) [reply app/routes/explore.js:87]: When has_image=null, this condition become true and only events with images are shown. 
When has_image=false, this condition hold false and all events whether image is present or not will be shown.

File: app/components/explore/side-bar.js (hunk 3 above)
Comment (sachinchauhan2889) [reply app/components/explore/side-bar.js:186]: Sir, i am not able to figure out what is wrong with identation. Could you please give hint or suggestion

File: app/routes/explore.js (hunk 5 above)
Comment (iamareebjamal) [reply app/routes/explore.js:87]: That's opposite logic though. has_image=true should only show events with images

Comment (sachinchauhan2889) [reply app/routes/explore.js:87]: > That's opposite logic though. has_image=true should only show events with images

Now i am little bit confused. 
Please correct me if i am wrong. 
Desired result:----> when has_image=null, all events shown whether it have image or not. 
When has_image=true, events with images must be shown.

Comment (iamareebjamal) [reply app/routes/explore.js:87]: Yes

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**